### PR TITLE
usage formatting

### PR DESCRIPTION
**Pull request by JyotikaGargg**

### CODE REVIEW

**Comment by stuartpa** [comment]
There is a bug for flags that do not have a short cut, they print out as just a "-":

  -, --version
      Print version information and exit

  -, --driver-logging-level
      Level of mssql driver messages to print.

**Comment by stuartpa** [comment]
Kubectl is doing a paragraph align of the descriptions, e.g.

```
Options:
    -c, --container='':
        Container name. If omitted, use the kubectl.kubernetes.io/default-container annotation for
        selecting the container to be attached or the first container in the pod will be chosen

    --no-preserve=false:
        The copied file/directory's ownership and permissions will not be preserved in the
        container
```

Where as this PR is wrapping to col 1, e.g.

```
  -G, --use-aad
      Tells sqlcmd to use Active Directory authentication. If no user name is provided, authentication method ActiveDirectoryDefault is used. If a password is provided, ActiveDirectoryPassword is used. Otherwise ActiveDirectoryInteractive is used.

  -E, --use-trusted-connection
      Uses a trusted connection instead of using a user name and password to sign in to SQL Server, ignoring any environment variables that define user name and password.

  -U, --user-name
      The login name or contained database user name.  For contained database users, you must provide the database name option
```

Can we align the description text as well please.

**Comment by stuartpa** [comment]
Given the sheer number of flags, it would be nice to be able to put the flags into groups.  It looks like Cobra doesn't support this yet, this PR would need to be merged:

https://github.com/spf13/cobra/pull/1778

Is there any other easy way to provide flag grouping that you know of?

**Comment by JyotikaGargg** [comment]
Updated the usage formatting similar to kubectl:
 -d, --database-name
      --This option sets the sqlcmd scripting variable
        SQLCMDDBNAME. This parameter specifies the initial
        database. The default is your login's
        default-database property. If the database does not
        exist, an error message is generated and sqlcmd
        exits.

  -X, --disable-cmd-and-warn
      --Disables commands that might compromise system security.
        Sqlcmd issues a warning and continues.

**Comment by JyotikaGargg** [comment]
> There is a bug for flags that do not have a short cut, they print out as just a "-":
> 
> -, --version Print version information and exit
> 
> -, --driver-logging-level Level of mssql driver messages to print.

Resolved this.

**Comment by stuartpa** [comment]
> > There is a bug for flags that do not have a short cut, they print out as just a "-":
> > -, --version Print version information and exit
> > -, --driver-logging-level Level of mssql driver messages to print.
> 
> Resolved this.

Looks much better:

  -c, --batch-terminator
      --Specifies the batch terminator. The default value is GO

  -s, --column-separator
      --Specifies the column separator character. Sets the
        SQLCMDCOLSEP variable.

There is an extra "--" at the beginning of each description...

**Comment by stuartpa** [comment]
Also, while you here, can you fix:

      --Specifies the SQL authentication method to use to connect
        to Azure SQL Database. One
    of:ActiveDirectoryDefault,ActiveDirectoryIntegrated,ActiveDirectoryPassword,ActiveDirectoryInteractive,ActiveDirectoryManagedIdentity,ActiveDirectoryServicePrincipal,SqlPassword
        
So there are some spaces in the string to it can wrap correctly.

**Comment by JyotikaGargg** [comment]
> Also, while you here, can you fix:
> 
> ```
>   --Specifies the SQL authentication method to use to connect
>     to Azure SQL Database. One
> of:ActiveDirectoryDefault,ActiveDirectoryIntegrated,ActiveDirectoryPassword,ActiveDirectoryInteractive,ActiveDirectoryManagedIdentity,ActiveDirectoryServicePrincipal,SqlPassword
> ```
> 
> So there are some spaces in the string to it can wrap correctly.

Resolved

**Comment by JyotikaGargg** [comment]
> > > There is a bug for flags that do not have a short cut, they print out as just a "-":
> > > -, --version Print version information and exit
> > > -, --driver-logging-level Level of mssql driver messages to print.
> > 
> > 
> > Resolved this.
> 
> Looks much better:
> 
> -c, --batch-terminator --Specifies the batch terminator. The default value is GO
> 
> -s, --column-separator --Specifies the column separator character. Sets the SQLCMDCOLSEP variable.
> 
> There is an extra "--" at the beginning of each description...

Resolved

**Comment by stuartpa** [comment]
This is looking much much better.  Nit: can you bring the help description back to spaces, so it is the same as kubectl, e.g.:

'''
Options:
    -c, --container='':
        Container name. If omitted, use the kubectl.kubernetes.io/default-container annotation for
        selecting the container to be attached or the first container in the pod will be chosen

    --no-preserve=false:
        The copied file/directory's ownership and permissions will not be preserved in the
        container
```

**Comment by JyotikaGargg** [comment]
> 
@stuartpa 
Are you talking about the description should look like below :

```
Flags:
-K,--application-intent
Declares the application workload type when connecting to a
server. The only currently supported value is ReadOnly. If
-K is not specified, the sqlcmd utility will not support
connectivity to a secondary replica in an Always On
availability group

      --authentication-method
Specifies the SQL authentication method to use to connect
to Azure SQL Database. One of: ActiveDirectoryDefault,
ActiveDirectoryIntegrated, ActiveDirectoryPassword,
ActiveDirectoryInteractive, ActiveDirectoryManagedIdentity,
ActiveDirectoryServicePrincipal, SqlPassword

-c,--batch-terminator
Specifies the batch terminator. The default value is GO
```

beacuse right now it looks like this :

```
Flags:
  -K, --application-intent
        Declares the application workload type when connecting to a
        server. The only currently supported value is
        ReadOnly. If -K is not specified, the sqlcmd
        utility will not support connectivity to a
        secondary replica in an Always On availability
        group

      --authentication-method
        Specifies the SQL authentication method to use to connect
        to Azure SQL Database. One of:
        ActiveDirectoryDefault, ActiveDirectoryIntegrated,
        ActiveDirectoryPassword,
        ActiveDirectoryInteractive,
        ActiveDirectoryManagedIdentity,
        ActiveDirectoryServicePrincipal, SqlPassword

  -c, --batch-terminator
        Specifies the batch terminator. The default value is GO

  -s, --column-separator
        Specifies the column separator character. Sets the
        SQLCMDCOLSEP variable.
```

I am liking lower one better as compared to first one. 
But still if we want to have flags without spaces , i could do those changes.

File: cmd/sqlcmd/sqlcmd.go [hunk 1]
```diff
@@ -162,11 +162,44 @@ func Execute(version string) {
 		},
 	}
 	setFlags(rootCmd, &args)
+	rootCmd.SetHelpFunc(func(cmd *cobra.Command, argss []string) {
+		fmt.Println(cmd.Long)
+		cmd.Flags().SetInterspersed(false)
+		fmt.Println("Flags:")
```
Comment (shueybubbles) [inline cmd/sqlcmd/sqlcmd.go:168]: >("Flags:") [](http://example.com/codeflow?start=13&length=10)

localizer

File: cmd/sqlcmd/sqlcmd.go [hunk 2]
```diff
@@ -162,11 +162,44 @@ func Execute(version string) {
 		},
 	}
 	setFlags(rootCmd, &args)
+	rootCmd.SetHelpFunc(func(cmd *cobra.Command, argss []string) {
+		fmt.Println(cmd.Long)
```
Comment (shueybubbles) [inline cmd/sqlcmd/sqlcmd.go:166]: >Println(cmd.Long) [](http://example.com/codeflow?start=6&length=17)

we should print the version too

I'm not certain about the copyright.

eg this is what ODBC sqlcmd has at the top

```
Microsoft (R) SQL Server Command Line Tool
Version 15.0.1300.359 NT
Copyright (C) 2019 Microsoft Corporation. All rights reserved.
```

**Comment by stuartpa** [comment]
> > 
> 
> @stuartpa Are you talking about the description should look like below :
> 
> ```
> Flags:
> -K,--application-intent
> Declares the application workload type when connecting to a
> server. The only currently supported value is ReadOnly. If
> -K is not specified, the sqlcmd utility will not support
> connectivity to a secondary replica in an Always On
> availability group
> 
>       --authentication-method
> Specifies the SQL authentication method to use to connect
> to Azure SQL Database. One of: ActiveDirectoryDefault,
> ActiveDirectoryIntegrated, ActiveDirectoryPassword,
> ActiveDirectoryInteractive, ActiveDirectoryManagedIdentity,
> ActiveDirectoryServicePrincipal, SqlPassword
> 
> -c,--batch-terminator
> Specifies the batch terminator. The default value is GO
> ```
> 
> beacuse right now it looks like this :
> 
> ```
> Flags:
>   -K, --application-intent
>         Declares the application workload type when connecting to a
>         server. The only currently supported value is
>         ReadOnly. If -K is not specified, the sqlcmd
>         utility will not support connectivity to a
>         secondary replica in an Always On availability
>         group
> 
>       --authentication-method
>         Specifies the SQL authentication method to use to connect
>         to Azure SQL Database. One of:
>         ActiveDirectoryDefault, ActiveDirectoryIntegrated,
>         ActiveDirectoryPassword,
>         ActiveDirectoryInteractive,
>         ActiveDirectoryManagedIdentity,
>         ActiveDirectoryServicePrincipal, SqlPassword
> 
>   -c, --batch-terminator
>         Specifies the batch terminator. The default value is GO
> 
>   -s, --column-separator
>         Specifies the column separator character. Sets the
>         SQLCMDCOLSEP variable.
> ```
> 
> I am liking lower one better as compared to first one. But still if we want to have flags without spaces , i could do those changes.

Like the lower one, but back two spaces

**Review by shueybubbles — CHANGES_REQUESTED** [review]
:clock1:

**Review by shueybubbles — APPROVED** [review]
:shipit:

**Pull request merged**